pkg/clusters/addons/kong/addon.go

```diff
@@ -190,6 +198,36 @@ func (a *Addon) Deploy(ctx context.Context, cluster clusters.Cluster) error {
 		return fmt.Errorf("%s: %w", stderr.String(), err)
 	}
 
+	// create a namespace ahead of deployment so things like license secrets and other configurations
+	// can be preloaded.
+	if err := clusters.CreateNamespace(ctx, cluster, a.namespace); err != nil {
+		return err
+	}
+
+	if a.proxyPullSecret != (pullSecret{}) {
+		// create the pull Secret
+		args := []string{"--kubeconfig", kubeconfig.Name(), "create", "secret", "-n",
+			a.namespace, "docker-registry", "proxy-pull",
+			"--docker-username", a.proxyPullSecret.Username, "--docker-password", a.proxyPullSecret.Password}
+		if a.proxyPullSecret.Server != "" {
+			args = append(args, []string{"--docker-server", a.proxyPullSecret.Server}...)
+		}
+		if a.proxyPullSecret.Email != "" {
+			args = append(args, []string{"--docker-email", a.proxyPullSecret.Email}...)
+		}
+		stderr = new(bytes.Buffer)
+		cmd = exec.CommandContext(ctx, "kubectl", args...)
+		cmd.Stdout = io.Discard
+		cmd.Stderr = stderr
+		if err := cmd.Run(); err != nil {
+			return fmt.Errorf("%s: %w", stderr.String(), err)
+		}
+		// use the pull Secret
+		a.deployArgs = append(a.deployArgs, []string{
+			"--set", "image.pullSecrets={proxy-pull}",
+		}...)
```

Comment:
Ideally we should be moving in a direction of reducing `kubectl` calls to make the framework more versatile. Since we're in a position to use `cluster.Client()` here to create the secret I think we should do so, rather than adding more `kubectl` calls.

Reply:
There is no escape from kubectl, only exec, and unfortunately it's not particularly amenable to re-using its helpers and requires everything effectively run as normal kubectl minus cobra :shrug: